test/expected/bgw_db_scheduler.out

```diff
@@ -804,3 +807,346 @@ SELECT * FROM bgw_log;
       0 |    200000 | test_job_4       | Execute job 4
 (7 rows)
 
+-- Test updating jobs list
+TRUNCATE bgw_log;
+SELECT _timescaledb_internal.stop_background_workers();
+ stop_background_workers 
+-------------------------
+ t
+(1 row)
+
+\c single :ROLE_SUPERUSER
+CREATE OR REPLACE FUNCTION ts_test_job_refresh() RETURNS TABLE(
+id INTEGER,
+application_name NAME,
+job_type NAME,
+schedule_interval INTERVAL,
+max_runtime INTERVAL,
+max_retries INT,
+retry_period INTERVAL,
+next_start TIMESTAMPTZ,
+timeout_at TIMESTAMPTZ,
+reserved_worker BOOLEAN,
+may_next_mark_end BOOLEAN
+)
+AS :MODULE_PATHNAME LANGUAGE C VOLATILE;
+CREATE FUNCTION verify_refresh_correct() RETURNS BOOLEAN LANGUAGE PLPGSQL AS
+$BODY$
+DECLARE
+    num_jobs INTEGER;
+    num_jobs_in_list INTEGER;
+BEGIN
+    SELECT COUNT(*) from _timescaledb_config.bgw_job INTO num_jobs;
+	select COUNT(*) from ts_test_job_refresh() JOIN _timescaledb_config.bgw_job USING (id,application_name,job_type,schedule_interval,max_runtime,max_retries,retry_period) INTO num_jobs_in_list;
```

Comment:
shouldn't we check this for correct updates as well, not just counts?

Reply:
But that's what the JOIN is? It will only join if the arguments match.

Reply:
👍